### PR TITLE
feat(fixtures): get plugin options for fixtures from options.json

### LINKED ISSUE #40: Feature request - Plugin options in fixtures directory
When using fixtures it would be nice if it was possible to provide plugin options in a `options.json` file. 
This would make it easier to test different features of a plugin, where different settings can be provided.

### PR DESCRIPTION
**Pull request by merceyz**
**What**: Made it possible to specify plugin options in a `options.json` file next to the fixtures.

**Why**: It's currently not possible to specify options for specific fixtures, or a group of fixtures.

**How**: Looks for a `options.json` file and if one is found loads it and passes it to the plugin, if the current directory doesn't contain a `code.js` file the settings are passed down the chain (inherited) instead.

Closes #40

### CODE REVIEW

**Comment by kwelch** [comment]
This looks like it makes sense. I would say moving forward with tests makes sense.

File: src/index.js [hunk 1]
```diff
@@ -209,16 +209,27 @@ function pluginTester({
...
     const tsCodePath = path.join(fixtureDir, 'code.ts')
     const blockTitle = caseName.split('-').join(' ')
     const codePath =
       (pathExists.sync(jsCodePath) && jsCodePath) ||
       (pathExists.sync(tsCodePath) && tsCodePath)
 
+    let fixturePluginOptions = {}
+    if (pathExists.sync(optionsPath)) {
+      fixturePluginOptions = JSON.parse(
+        fs.readFileSync(optionsPath, 'utf8').toString(),
```
Comment (kentcdodds) [inline src/index.js]: Any reason we're not just using `require(optionsPath)` here?

Comment (merceyz) [reply src/index.js]: Not really, is that preferred?

Comment (kentcdodds) [reply src/index.js]: Meh, I'm fine either way. I just think require looks nicer.

**Comment by merceyz** [comment]
Had to fix a few issues to get the tests to pass

**Comment by merceyz** [comment]
@kentcdodds Docs updated

**Pull request merged**